constraints/src/main/java/gov/nasa/jpl/aerie/constraints/json/ConstraintParsers.java

```diff
@@ -462,7 +462,7 @@ static JsonParser<ForEachActivitySpans> forEachActivitySpansF(final JsonParser<E
         .field("expression", spansExpressionP)
         .map(
             untuple((kind, actType, alias, expression) -> new ForEachActivitySpans(actType, alias, expression)),
```

Comment:
Musing - would it make any sense to expand the constraints' `ForEachActivity` nodes to filter by parameters as well?

I don't suppose it adds any expressive power - since you can already refer to parameters in the body of a constraint.

Reply:
Currently this filtering option isn't exposed to the constraints eDSL for either `Spans` or `Constraint`. It could be added and would dramatically simplify this pattern, but I'd say that's beyond the scope of this ticket. At a minimum we'd probably have to move `ActivityExpression` to the constraints package, and I would put money on that being harder than it sounds.